### PR TITLE
Use only pluginZip publication of Apache Maven artifacts

### PR DESCRIPTION
**Pull request by reta**
### Description
Use only pluginZip publication of Apache Maven artifacts

### Issues Resolved
Fixes https://github.com/opensearch-project/flow-framework/issues/186

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
@reta still not able to see artifacts for `opensearch-flow-framework` under `m2`

```
ls ~/.m2/repository/org/opensearch/opensearch
opensearch/                                              opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha1    opensearch-ml/
opensearch-benchmarks/                                   opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha256  opensearch-ml-client/
opensearch-cli/                                          opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.sha512  opensearch-ml-common/
opensearch-common/                                       opensearch-job-scheduler-spi-1.2.0.0-sources.jar         opensearch-ml-spi/
opensearch-compress/                                     opensearch-job-scheduler-spi-1.2.0.0-sources.jar.md5     opensearch-nio/
opensearch-core/                                         opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha1    opensearch-performance-analyzer/
opensearch-dissect/                                      opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha256  opensearch-plugin-classloader/
opensearch-geo/                                          opensearch-job-scheduler-spi-1.2.0.0-sources.jar.sha512  opensearch-plugin-cli/
opensearch-grok/                                         opensearch-job-scheduler-spi-1.2.0.0.jar                 opensearch-secure-sm/
opensearch-job-scheduler/                                opensearch-job-scheduler-spi-1.2.0.0.jar.md5             opensearch-ssl-config/
opensearch-job-scheduler-sample-extension/               opensearch-job-scheduler-spi-1.2.0.0.jar.sha1            opensearch-telemetry/
opensearch-job-scheduler-spi/                            opensearch-job-scheduler-spi-1.2.0.0.jar.sha256          opensearch-upgrade-cli/
opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar         opensearch-job-scheduler-spi-1.2.0.0.jar.sha512          opensearch-x-content/
opensearch-job-scheduler-spi-1.2.0.0-javadoc.jar.md5     opensearch-launchers/  
```

**Comment by reta** [comment]
> @reta Thanks for the PR. Not to be a bad news here. Still not able to see artifacts for `opensearch-flow-framework` under `m2`

@owaiskazi19 Please check `ls ~/.m2/repository/org/opensearch/plugin`

**Comment by owaiskazi19** [comment]
> > @reta Thanks for the PR. Not to be a bad news here. Still not able to see artifacts for `opensearch-flow-framework` under `m2`
> 
> @owaiskazi19 Please check `ls ~/.m2/repository/org/opensearch/plugin` , since this is plugin

Ah! Yes. Missed that. First time working with a plugin.

**Comment by owaiskazi19** [comment]
@reta we have spotless for our build files too. Can you run `./gradlew spotlessApply` and push again? Thanks

File: build.gradle [hunk 1]
```diff
@@ -261,3 +268,13 @@ diffCoverageReport {
         failOnViolation = true
     }
 }
+
+
+tasks.withType(AbstractPublishToMaven) {
```
Comment (amitgalitz) [inline build.gradle:273]: just wondering why does no other plugin need this?

**Review by amitgalitz — APPROVED** [review]
LGTM, just wanted to get an understanding if known what is different in our plugin that needs this since I didn't have context on the changes needed here

File: build.gradle [hunk 2]
```diff
@@ -86,23 +88,28 @@ publishing {
...
+                    developer.appendNode('url', 'https://github.com/opensearch-project/flow-framework')
                 }
             }
         }
     }
 
     repositories {
+        maven {
+            name = 'staging'
+            url = "${rootProject.buildDir}/local-staging-repo"
```
[the remark this answers is not in the document]
Comment (reta) [reply build.gradle:111]: yes

File: build.gradle (hunk 1 above)
Comment (owaiskazi19) [reply build.gradle:273]: This is what I missed in my PR which fixed the publishing. @reta would be a better person to answer this.

File: build.gradle [hunk 3]
```diff
@@ -35,9 +35,6 @@ dependencyLicenses.enabled = false
 loggerUsageCheck.enabled = false
 thirdPartyAudit.enabled = false
 
-// No need to validate pom, as we do not upload to maven/sonatype
-validateNebulaPom.enabled = false
```
Comment (owaiskazi19) [inline build.gradle:39]: or add this line back?

Comment (reta) [reply build.gradle]: > or add this line back?

Yes, added it back, thanks!

**Pull request merged**